Source/ACE/Entity/WorldObject.cs

```diff
@@ -160,101 +160,101 @@ public virtual void SerializeCreateObject(BinaryWriter writer)
             if ((WeenieFlags & WeenieHeaderFlag.PuralName) != 0)
                 writer.WriteString16L(GameData.NamePlural);
 
-            if ((WeenieFlags & WeenieHeaderFlag.ItemCapacity) != 0)
-                writer.Write(GameData.ItemCapacity);
+            if (((WeenieFlags & WeenieHeaderFlag.ItemCapacity) != 0) && GameData.ItemCapacity != null)
```

Comment:
You'll have to apply this to all of these.  The basic issue is that we have flags that are set in bulk.  **Flags should be calculated based on the presence and absence of data, not arbitrarily set and then validated after being written**.

```cs
if ((WeenieFlags & WeenieHeaderFlag.ItemCapacity) != 0)
{
    if (GameData.ItemCapacity != null)
    {
        // write out items
    }
    else
    {
        // we wrote the flag in the header, failing to write at least SOMETHIGN will cause the client to barf on the packet.
        // let's try writing a 0 to test it
       writer.Write((uint)0);
    }
}